pkg/cache/upstream/cache.go

```diff
@@ -52,7 +52,7 @@ type Cache struct {
 	publicKeys []signature.PublicKey
 }
 
-func New(ctx context.Context, u *url.URL, pubKeys []string) (Cache, error) {
+func New(ctx context.Context, u *url.URL, pubKeys []string, priority int64) (Cache, error) {
```

Comment:
Why not have the parsePriority passes it from the url instead of passing it from main?

Reply:
Ah, very good point - much cleaner!